census/estimator/trainer/task.py

```diff
@@ -0,0 +1,162 @@
+import argparse
+import json
+import os
+
+import model
+
+from tensorflow.contrib.learn.python.learn import learn_runner
+from tensorflow.contrib.learn.python.learn.utils import (
+    saved_model_export_utils)
+from tensorflow.contrib import metrics, learn
+
+
+def generate_experiment_fn(train_file,
+                           eval_file,
+                           num_epochs=None,
+                           train_batch_size=40,
+                           eval_batch_size=40,
+                           embedding_size=8,
+                           serving_input_fn_type='JSON',
+                           hidden_units=None,
+                           **experiment_args):
+  def _experiment_fn(output_dir):
+    train_input = model.generate_input_fn(
+        train_file, num_epochs=num_epochs, batch_size=train_batch_size)
+    eval_input = model.generate_input_fn(
+        eval_file,
+        batch_size=eval_batch_size,
+        skip_header_lines=1,
+        shuffle=False
+    )
+    if args.serving_input_fn_type == 'JSON':
```

Comment:
Yes but it's requires constructing the CSV string on the client side, which introduces lots of potential problems. Also, AFAICT `SavedModel` + `Estimator` doesn't support single input tensors, so you'll need to tag each input tensor with a key regardless. So instead of it being CSV, it will be JSON where one of the values is a CSV string. Which is super ugly.